### ISSUE TITLE
Add a minimum width setting for editor docks

### ISSUE THREAD

**Issue by TheDuriel** [godotengine/godot-proposals#5685]
### Describe the project you are working on

Various, this relates to using the Godot editor.

### Describe the problem or limitation you are having in your project

Currently, Editor Docks initialize with a width depending on their contents. By default this is around 200 pixels. The width can then be adjusted by the user to suit their needs.

The problem is that, many docks initialize at a width that is too narrow to be usable. For example, if I instance a Control, and a Child MenuButton. The name of the MenuButton node in the Scene dock is already too long for the default width. Similarly, many of the properties displayed in the inspector will spill into the inspector scroll bar (this is a bug), or be clipped and unable to be read until the inspector is expanded.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a new Editor Setting that allows users to set a Minimum Width for Editor Docks. Specifically Godots default docks like, FileSystem, Scene, Inspector, etc.

Set a reasonable default value for 1080p displays, (perhaps a fallback for the rare 720p user could be added as well.)

I propose 256 or at least 224. (Slightly larger than the inspectors natural minimum size.) Which is a value I used for years in a very wonky Editor Plugin.

For easy of use an implementation, this setting would not actually affect the Dock itself. **But the node containing the Dock.**

Additionally this will make the Editor appear more consistent and make me happy not having to look at every column being a difference size and changing size when I change tabs all the time.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

You can write an absolutely atrocious plugin like this one here: https://gist.github.com/TheDuriel/cc5d0edd480f4acabbd5ae5aff27bce3 Which includes a few other comfort features. But you can see how silly it is. (There are better ways to get the docks. But geeeez.)

### Is there a reason why this should be core and not an add-on in the asset library?

Editor feature.

**Comment by KoBeWi** [comment]
You can manually adjust the width of the docks and they will not shrink past that size (effectively acting as minimum width) and only expand when necessary. It used to be unreliable, but it was recently changed in 4.0. The only problem is that this has to be set per-project and the default width is rather small.

**Comment by TheDuriel** [comment]
I'd still vastly prefer uniformity between columns and consistency between projects. Hence either a setting, or just a reasonable default.

**Comment by YuriSizov** [comment]
You can save dock layouts, so the consistency between projects part is already covered, even if a bit convoluted.

**Comment by TheDuriel** [comment]
There is no means to set precise values. And someone opening up the Engine for the first time is still greeted with unusably narrow docks.

Like my proposal stated. You need **two** nodes in the scene browser to be unable to read the second nodes name fully. Not accounting for the various Icons for scripts and locking that may be added on to such a node.

And it's fairly well understood in UI design that users will rarely deviate from default layouts and options to "fix" these issues, even if a single click away.

So I could instead make an issue about a usability breaking bug? "Unable to read node and property names with default Editor Layout."

**Comment by YuriSizov** [comment]
Precise values are not important after you've tuned it to your liking. But you are correct otherwise.

Problem now is, narrow columns on a bigger resolution are a nuisance, while wide columns on a smaller resolution may be unusable and unfixable if they result in an overflow. We are consciously putting effort to make the UI of the editor work even on a lower end laptop screen. So to preserve this usability goal, we would need to somehow adjust the defaults based on the screen resolution during the first load.

Also would be useful to have an editor-wide setting for a default layout, so you could indeed set it and forget it for all future projects on the same machine.

**Comment by TheDuriel** [comment]
> Precise values are not important after you've tuned it to your liking.

It is my liking for them to be precise :P And I can not currently achieve that without a hacky editor plugin.

> Problem now is...

Absolutely. I am proposing we set a reasonable default for 1080p displays. Which may be ignored if the aspect ratio or resolution drops below a certain threshold. Or we can ask those users to change it in the editor settings. I do come at this from the perspective of someone who regularly uses Godot on such a laptop. (Tiny Dell Latitude that tries to be a Thinkpad)

> Also would be useful to have an editor-wide setting for a default layout,

Very agreed! There might be a proposal or old issue for this somewhere.

There is another comment to be made about rearranging the Docks to follow expectations from common creator tools, but that's a whole different thing I'll not breach here further. One step at a time.

**Comment by wlwl2** [comment]
@TheDuriel I agree. Editor > Editor Settings... > Interface > Editor > Minimum Width For **All** Editor Docks
